packages/frontend/src/components/EditorSettings/FlowTransfer/TransferFlowModal.tsx

```diff
@@ -0,0 +1,82 @@
+import { useCallback } from 'react'
+import { useParams } from 'react-router-dom'
+import { useMutation } from '@apollo/client'
+import {
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from '@chakra-ui/react'
+import { Button, useToast } from '@opengovsg/design-system-react'
+import { CREATE_FLOW_TRANSFER } from 'graphql/mutations/create-flow-transfer'
+import { GET_PENDING_FLOW_TRANSFER } from 'graphql/queries/get-pending-flow-transfer'
+
+interface TransferFlowModalProps {
+  onClose: () => void
+  newOwnerEmail: string
+}
+
+export default function TransferFlowModal(props: TransferFlowModalProps) {
+  const { onClose, newOwnerEmail } = props
+  const { flowId } = useParams()
+
+  const [createFlowTransfer] = useMutation(CREATE_FLOW_TRANSFER)
+  const toast = useToast()
+
+  const onFlowTransferCreate = useCallback(async () => {
+    onClose()
+    await createFlowTransfer({
+      variables: {
+        input: {
+          flowId,
+          newOwnerEmail,
+        },
+      },
+      refetchQueries: [GET_PENDING_FLOW_TRANSFER],
+      onCompleted: () => {
+        toast({
+          title:
+            'Transfer has been requested. Please get the new owner to login and approve!',
```

Comment:
nit: stacey ok with the copy? I thought she usually dont like exclamation marks

Reply:
yea cus she never gave me the figma for modals, ill consult her again